mmseg/models/backbones/twins.py

```diff
@@ -0,0 +1,646 @@
+                 num_classes=1000,
+                 embed_dims=[64, 128, 256, 512],
+                 num_heads=[1, 2, 4, 8],
+                 mlp_ratios=[4, 4, 4, 4],
+                 out_indices=(0, 1, 2, 3),
+                 qkv_bias=False,
+                 drop_rate=0.,
+                 attn_drop_rate=0.,
+                 drop_path_rate=0.,
+                 norm_cfg=dict(type='LN'),
+                 depths=[3, 4, 6, 3],
+                 sr_ratios=[8, 4, 2, 1],
+                 extra_norm=False,
+                 init_cfg=None):
+        super(PCPVT, self).__init__(init_cfg=init_cfg)
+        self.num_classes = num_classes
+        self.depths = depths
+
+        # patch_embed
+        self.patch_embeds = ModuleList()
+        self.pos_drops = ModuleList()
+        self.blocks = ModuleList()
+
+        for i in range(len(depths)):
+            if i == 0:
+                input_size = img_size
+                self.patch_embeds.append(
+                    PatchEmbed(
+                        in_channels=in_channels,
+                        embed_dims=embed_dims[i],
+                        conv_type='Conv2d',
+                        kernel_size=patch_size,
+                        stride=patch_size,
+                        padding='corner',
+                        norm_cfg=norm_cfg,
+                        input_size=input_size,
+                        init_cfg=None))
+            else:
+                patch_size = 2
+                input_size = img_size // patch_size // 2**(i - 1)
+                self.patch_embeds.append(
+                    PatchEmbed(
+                        in_channels=embed_dims[i - 1],
+                        embed_dims=embed_dims[i],
+                        conv_type='Conv2d',
+                        kernel_size=patch_size,
+                        stride=patch_size,
+                        padding='corner',
+                        norm_cfg=norm_cfg,
+                        input_size=input_size,
+                        init_cfg=None))
+
+            self.pos_drops.append(nn.Dropout(p=drop_rate))
+
+        # transformer encoder
+        dpr = [
+            x.item() for x in torch.linspace(0, drop_path_rate, sum(depths))
+        ]  # stochastic depth decay rule
+        cur = 0
+
+        for k in range(len(depths)):
+            _block = ModuleList([
+                GSAEncoderLayer(
+                    embed_dims=embed_dims[k],
+                    num_heads=num_heads[k],
+                    feedforward_channels=mlp_ratios[k] * embed_dims[k],
+                    attn_drop_rate=attn_drop_rate,
+                    drop_rate=drop_rate,
+                    drop_path_rate=dpr[cur + i],
+                    num_fcs=2,
+                    qkv_bias=qkv_bias,
+                    act_cfg=dict(type='GELU'),
+                    norm_cfg=dict(type='LN'),
+                    sr_ratio=sr_ratios[k]) for i in range(depths[k])
+            ])
+            self.blocks.append(_block)
```

Comment:
```suggestion
            self.layers.append(_block)
```
use layers or stages.